### PR TITLE
feat(efs): support for new `elastic` throughputmode

### PR DESCRIPTION
**Pull request by smsruthi**
----

### All Submissions:

* [ yes] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Construct Runtime Dependencies:

* [ ] This PR adds new construct runtime dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-construct-runtime-dependencies)

### New Features

* [ ] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [ ] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by mrgrain** [comment]
Hey @smsruthi thanks for the PR!

I've exempted the integration test as I don't think we need one here.

However, would you mind updating the README: https://github.com/aws/aws-cdk/tree/5a20554754beda761623cbd9c71934ab89ec2ff2/packages/%40aws-cdk/aws-efs#create-an-amazon-efs-file-system

Particular the part about replacement seems out of date now. (see [Restrictions on switching throughput modes and changing provisioned amount](https://docs.aws.amazon.com/efs/latest/ug/performance.html#throughput-modes))

Additionally, with "Elastic Throughput mode is available only for file systems that are configured with the General Purpose performance mode."  we need to add some validation for this constraint. Similar to the one we have for [PROVISIONED](https://github.com/aws/aws-cdk/blob/5a20554754beda761623cbd9c71934ab89ec2ff2/packages/@aws-cdk/aws-efs/lib/efs-file-system.ts#L343).

Lastly, we also need unit tests covering the validation cases.

**Comment by sbyooki** [comment]
- You still need to put comma at the end of
```
performanceMode: PerformanceMode.MAX_IO
```
If fails the build

- A test failed with this:
- ```
 Expected pattern: /PerformanceMode cannot be MaxIO when using ThroughputMode ELASTIC/
    Received message: "Property performanceMode cannot be MAX_IO when throughputMode is ELASTIC"
```
^^ Other than making the two strings the same, I suggest you change the language more towards ELASTIC ThroughputMode is not supported for file systems with PerformanceMode of MaxIO

File: packages/@aws-cdk/aws-efs/test/efs-file-system.test.ts [hunk 1]
```diff
@@ -162,6 +162,29 @@ test('file system is created correctly with bursting throughput mode', () => {
...
+});
+
+test('Exception when throughput mode is set to ELASTIC, performance mode cannot be MaxIO', () => {
+  expect(() => {
+    new FileSystem(stack, 'EfsFileSystem', {
+      vpc,
+      throughputMode: ThroughputMode.ELASTIC,
+      performanceMode: PerformanceMode.MAX_IO,
+    });
+  }).toThrowError(/Elastic ThroughputMode is not supported for file systems with PerformanceMode of MaxIO/);
```
Comment (mrgrain) [inline packages/@aws-cdk/aws-efs/test/efs-file-system.test.ts:185]:
```suggestion
  }).toThrowError(/ThroughputMode ELASTIC is not supported for file systems with performanceMode MAX_IO/);
```

File: packages/@aws-cdk/aws-efs/lib/efs-file-system.ts [hunk 2]
```diff
@@ -333,6 +344,9 @@ export class FileSystem extends FileSystemBase {
       throw new Error('Property provisionedThroughputPerSecond is required when throughputMode is PROVISIONED');
     }
 
+    if (props.throughputMode === ThroughputMode.ELASTIC && props.performanceMode === PerformanceMode.MAX_IO) {
+      throw new Error('Elastic ThroughputMode is not supported for file systems with PerformanceMode of MaxIO');
```
Comment (mrgrain) [inline packages/@aws-cdk/aws-efs/lib/efs-file-system.ts:348]:
```suggestion
      throw new Error('ThroughputMode ELASTIC is not supported for file systems with performanceMode MAX_IO');
```

**Review by mrgrain — CHANGES_REQUESTED** [review]
Looks good now, thank you. I'll take care of the text changes.

File: packages/@aws-cdk/aws-efs/README.md [hunk 3]
```diff
@@ -40,6 +40,8 @@ const fileSystem = new efs.FileSystem(this, 'MyEfsFileSystem', {
 });
 ```
 
+⚠️ An Amazon EFS file system's performance mode can't be MaxIo while its throughputMode is ELASTIC.
```
Comment (mrgrain) [inline packages/@aws-cdk/aws-efs/README.md:43]:
```suggestion
⚠️ An Amazon EFS file system's performance mode can't be MAX_IO when its throughputMode is ELASTIC.
```

**Review by mrgrain — APPROVED** [review]

**Pull request merged**